File: .skills/openclaw-skills/skills/chenghaifeng08-creator/self-evolve-automaton/scripts/rollback.js
#!/usr/bin/env node
/**
 * Self-Evolve Rollback Script
 * 
 * Restores the most recent backup of a file.
 * Usage: node rollback.js <file-name>
 */

const fs = require('fs');
const path = require('path');

const WORKSPACE = 'C:\\Users\\Administrator\\.openclaw\\workspace';
const BACKUP_DIR = path.join(WORKSPACE, 'backups', 'self-evolve');

// Check if backup directory exists
if (!fs.existsSync(BACKUP_DIR)) {
  console.error('❌ Error: Backup directory not found');
  process.exit(1);
}

// Get file name from argument
const fileName = process.argv[2];

if (!fileName) {
  // List available backups
  console.log('Available backups:\n');
  const files = fs.readdirSync(BACKUP_DIR)
    .filter(f => f.endsWith('.bak'))
    .sort()
    .reverse();
  
  files.slice(0, 10).forEach((f, i) => {
    console.log(`  ${i + 1}. ${f}`);
  });
  
  if (files.length > 10) {
    console.log(`  ... and ${files.length - 10} more`);
  }
  
  console.log('\nUsage: node rollback.js <file-name>');
  process.exit(0);
}

// Find most recent backup for this file
const backups = fs.readdirSync(BACKUP_DIR)
  .filter(f => f.startsWith(fileName + '.') && f.endsWith('.bak'))
  .sort()
  .reverse();

if (backups.length === 0) {
  console.error(`❌ No backups found for: ${fileName}`);
  process.exit(1);
}

const latestBackup = backups[0];
const backupPath = path.join(BACKUP_DIR, latestBackup);
const originalPath = path.join(WORKSPACE, fileName);

// Confirm rollback
console.log(`Rolling back:`);
console.log(`  File: ${fileName}`);
console.log(`  From: ${latestBackup}`);
console.log(`  To: ${originalPath}`);

try {
  fs.copyFileSync(backupPath, originalPath);
  console.log('✓ Rollback complete');
} catch (e) {
  console.error(`❌ Error: ${e.message}`);
  process.exit(1);
}

File: .skills/openclaw-skills/skills/guanqi0914/github-hosts-optimizer-cn/scripts/rollback.py
#!/usr/bin/env python3
"""GitHub Hosts Optimizer - 回滚脚本"""
import sys
from pathlib import Path
from datetime import datetime

BACKUP_DIR = Path("/etc")

def find_latest_backup():
    backups = sorted(BACKUP_DIR.glob("hosts.backup_*"), reverse=True)
    if backups:
        return backups[0]
    return None

def rollback():
    backup = find_latest_backup()
    if not backup:
        print("❌ 未找到备份文件")
        sys.exit(1)
    
    print(f"回滚到: {backup}")
    content = backup.read_text()
    Path("/etc/hosts").write_text(content)
    print(f"✅ /etc/hosts 已回滚")
    print(f"   备份仍保留在: {backup}")

if __name__ == "__main__":
    rollback()
